githubapp/client_creator.go

```diff
@@ -229,6 +283,66 @@ func applyMiddleware(base *http.Client, middleware []ClientMiddleware) {
 	}
 }
 
+func newAppInstallation(integrationID int, privKeyBytes []byte, v3BaseURL string) (ClientMiddleware, error) {
+	var transportError error
+	installation := func(next http.RoundTripper) http.RoundTripper {
+		itr, err := ghinstallation.NewAppsTransport(next, integrationID, privKeyBytes)
+		if err != nil {
+			transportError = err
+			return next
+		}
+		// leaving the v3 URL since this is used to refresh the token, not make queries
+		itr.BaseURL = strings.TrimSuffix(v3BaseURL, "/")
+		return itr
+	}
+	return installation, transportError
```

Comment:
This still won't work as expected - `transportError` is nil at the time of the return and is returned by value, so future modifications won't be reflected to the caller.

That said, returning an `*error` and then checking if `*err != nil` in the client should work: https://play.golang.org/p/611tCTlwhWs